### PR TITLE
OAK-9699 Document Elastic Indexes

### PR DESCRIPTION
**Pull request by thomasmueller**
A first step.

### CODE REVIEW

File: oak-doc/src/site/markdown/query/elastic.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+  -->
+
+## Elastic Index
+
+Oak supports Elasticsearch (Elastic for short) based indexes to support 
+both property constraint and full text constraints. 
+Elastic indexes support similar features than [Lucene][lucene] indexes,
```
Comment (nit0906) [inline oak-doc/src/site/markdown/query/elastic.md]: similar features **than** [Lucene][lucene] --> similar features **as** [Lucene][lucene]

File: oak-doc/src/site/markdown/query/elastic.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  -->
+
+## Elastic Index
+
+Oak supports Elasticsearch (Elastic for short) based indexes to support 
+both property constraint and full text constraints. 
+Elastic indexes support similar features than [Lucene][lucene] indexes, 
+however there are differences:
+
+* The `type` is `elastic`.
```
Comment (nit0906) [inline oak-doc/src/site/markdown/query/elastic.md]: elasticsearch instead of elastic.

File: oak-doc/src/site/markdown/query/elastic.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+* The index definition needs to be under `/oak:index`.
+  Other locations are not supported.
+* The `async` property needs to be set to `async`.
```
Comment (nit0906) [inline oak-doc/src/site/markdown/query/elastic.md]: The `async` property needs to be set to `async`.  --> The `async` property needs to be set to `elastic-async`.

File: oak-doc/src/site/markdown/query/elastic.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  Other locations are not supported.
+* The `async` property needs to be set to `async`. 
+  Synchronous, `nrt` or other lanes are not supported.
+  Indexes are updated asynchronously.
+* `refresh` is ignored.
+  Changes take effect immediately after changing them.
+  Existing documents in Elasticsearch are not changed.
+* `reindex` is ignored.
+  Indexes are automatically built when needed.
+  We recommend to build then using the `oak-run` tool.
```
Comment (nit0906) [inline oak-doc/src/site/markdown/query/elastic.md]: build them*

**Review by nit0906 — CHANGES_REQUESTED** [review]
Some minor comments. 

Also , 
Before making this public - should we remove https://github.com/apache/jackrabbit-oak/tree/1.22/oak-search-elastic from the 1.22 branch ?
It’s an incomplete implementation in this branch - since we might have started when this was trunk, but never really backported anything post 1.22 was cut off.

**Review by fabriziofortino — COMMENTED** [review]
Should we add something around `dynamicBoost`?

There are no differences from a configuration point of view but the behavior changes. In general, with Elastic dynamically boosted terms are queryable through lucene syntax.

Perhaps, for dynamicBoost we could provide a separate section that describes the differences with Lucene, including dynamicBoostLite.

File: oak-doc/src/site/markdown/query/elastic.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+  -->
+
+## Elastic Index
+
+Oak supports Elasticsearch (Elastic for short) based indexes to support 
+both property constraint and full text constraints.
```
Comment (chibulcuteanu) [inline oak-doc/src/site/markdown/query/elastic.md]:
```suggestion
Oak supports Elasticsearch (Elastic for short) based indexes for both property constraint and full text constraints. 
```

File: oak-doc/src/site/markdown/query/elastic.md (hunk 4 above)
Comment (chibulcuteanu) [inline oak-doc/src/site/markdown/query/elastic.md]:
```suggestion
  Indexes are NOT automatically built when needed. They can be build by setting the `reindex` property to `true` or by using the `oak-run` tool.
  We recommend to build them using the `oak-run` tool.
```

File: oak-doc/src/site/markdown/query/elastic.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+* The `type` is `elastic`.
+* The index definition needs to be under `/oak:index`.
+  Other locations are not supported.
+* The `async` property needs to be set to `async`. 
+  Synchronous, `nrt` or other lanes are not supported.
+  Indexes are updated asynchronously.
+* `refresh` is ignored.
+  Changes take effect immediately after changing them.
+  Existing documents in Elasticsearch are not changed.
+* `reindex` is ignored.
```
Comment (chibulcuteanu) [inline oak-doc/src/site/markdown/query/elastic.md]: what do you mean this is ignored? It's the only way to build an ES index after deploying the index definition directly on the repository (no oak-run).

Comment (thomasmueller) [reply oak-doc/src/site/markdown/query/elastic.md]: Ah sorry I thought it's ignored. Will remove it.

File: oak-doc/src/site/markdown/query/elastic.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+* The index definition needs to be under `/oak:index`.
+  Other locations are not supported.
+* The `async` property needs to be set to `elastic-async`. 
+  Synchronous, `nrt` or other lanes are not supported.
+  Indexes are updated asynchronously.
+* `refresh` is ignored.
+  Changes take effect immediately after changing them.
+  Existing documents in Elasticsearch are not changed.
+* Indexes are NOT automatically built when needed: 
+  They can be build by setting the `reindex` property to `true` or by using the `oak-run` tool.
```
Comment (chibulcuteanu) [inline oak-doc/src/site/markdown/query/elastic.md]: typo
```suggestion
  They can be built by setting the `reindex` property to `true` or by using the `oak-run` tool.
```

**Review by chibulcuteanu — APPROVED** [review]

**Review by nit0906 — APPROVED** [review]

**Review by fabriziofortino — APPROVED** [review]

**Pull request merged**